app/javascript/components/editor/TestSummary.tsx

```diff
@@ -44,19 +47,19 @@ export function TestSummary({
         {isPresent(test.testCode) ? (
           <div className="--info">
             <h3>Code Run</h3>
-            <pre>{test.testCode}</pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.testCodeHtml }}></pre>
           </div>
         ) : null}
         {isPresent(test.message) ? (
           <div className="--info">
             <h3>{messageLabels[test.status]}</h3>
-            <pre>{test.message}</pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.messageHtml }}></pre>
           </div>
         ) : null}
         {isPresent(test.output) ? (
           <div className="--info">
             <h3>Your Output</h3>
-            <pre>{test.output}</pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.outputHtml }}></pre>
```

Comment:
I've kept the output as HTML, as student could theoretically use colors in their output, which the test runner could capture.